### PR TITLE
chore: update samples to 4.30

### PR DESCRIPTION
**Pull request by andygup**
Notable repo changes:
- Prep for the addition of components samples
- @arcgis/core samples folder renamed from `esm-samples` to `core-samples`
- Reverted webmap change to use the old/previous webmap. We now have a duplicate copy.
- Added download links for the individual sample folders, so you don't have to clone the whole repo.

### CODE REVIEW

File: core-samples/jsapi-custom-ui/package.json [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
   },
   "devDependencies": {
-    "@types/react": "^18.2.57",
-    "@types/react-dom": "^18.2.19",
-    "@vitejs/plugin-react": "^4.2.1",
-    "vite": "^5.1.4"
+    "@types/react": "^18.3.3",
+    "@types/react-dom": "^18.3.0",
+    "@vitejs/plugin-react": "^4.3.0",
+    "vite": "^5.2.12"
```
Comment (dasa) [inline core-samples/jsapi-custom-ui/package.json:19]: Use latest?

**Review by dasa — APPROVED** [review]
🥳

File: core-samples/jsapi-custom-ui/package.json (hunk 1 above)
Comment (andygup) [reply core-samples/jsapi-custom-ui/package.json]: Updated, thanks:

```json
    "@vitejs/plugin-react": "^4.3.1",
    "vite": "^5.3.1"
```

File: core-samples/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+The on-disk build output metrics represent files created by the local build and written to disk. At runtime, your application only loads a portion of the bundles depending on which `@arcgis/core` modules are used in your project. The number and size of bundles is dependant on multiple factors and will vary based on framework, module bundler, transpiling and related configurations. The compile process can output several hundred or more bundles on-disk due to the ArcGIS JavaScript Maps SDK using dynamic imports. 
+
+The performance indicators will vary depending on the device or virtual machine, and they are best used for analyzing trends over time. They are dependant on many hardware and software factors including CPU load, memory usage and internet speed during the time window when they are captured. For example, the numbers posted in the `metrics` directory come from running a cloud-based GitHub action. It's very likely that the metrics will be different from when you capture metrics locally on your development machine in a headless environment, or directly in a browser developer console.
```
Comment (sagewall) [inline core-samples/README.md:48]:
```suggestion
The performance indicators will vary depending on the device or virtual machine, and they are best used for analyzing trends over time. They depend on many hardware and software factors including CPU load, memory usage and internet speed during the time window when they are captured. For example, the numbers posted in the `metrics` directory come from running a cloud-based GitHub action. It's very likely that the metrics will be different from when you capture metrics locally on your development machine in a headless environment, or directly in a browser developer console.
```

Comment (sagewall) [inline core-samples/README.md:48]:
```suggestion
The performance indicators will vary depending on the device or virtual machine, and they are best used for analyzing trends over time. They depend on many hardware and software factors including CPU load, memory usage and internet speed during the time window when they are captured. For example, the numbers posted in the `metrics` directory come from running a cloud-based GitHub action. It's very likely that the metrics will be different from when you capture metrics locally on your development machine in a headless environment, or directly in a browser developer console.
```

**Review by sagewall — APPROVED** [review]
Looks great, found a single spelling error (if your not British) `dependant` and then I thought it could just be shortened to depend

**Review by odoe — APPROVED** [review]

**Comment by andygup** [comment]
Making updates to the github action. There will be a few extra build email notifications.

**Comment by andygup** [comment]
`analyze-builds` github updates:
- actions/checkout@v4 - fixes deprecation warning for @v4
- puppeteer v22
- express v4.19

**Pull request merged**